packages/react-components/react-divider/src/components/Divider/useDivider.ts

```diff
@@ -25,13 +25,16 @@ export const useDivider_unstable = (props: DividerProps, ref: React.Ref<HTMLElem
     },
 
     root: slot.always(
-      getNativeElementProps('div', {
+      getIntrinsicElementProps('div', {
         role: 'separator',
         'aria-orientation': vertical ? 'vertical' : 'horizontal',
         'aria-labelledby': props.children ? dividerId : undefined,
         ...props,
-        ref,
-      }),
+        // FIXME:
+        // `ref` is wrongly assigned to be `HTMLElement` instead of `HTMLDivElement`
+        // but since it would be a breaking change to fix it, we are casting ref to it's proper type
+        ref: ref as React.Ref<HTMLDivElement>,
+      } as const),
```

Comment:
Why do you need to cast everything as `const` here?

Reply:
Because `aria-orientation` type is `'vertical' | 'horizontal'` and without const assertion TS will assume we're delivering `string` instead of `'vertical' | 'horizontal'`, which is a mistake that now we can get thanks to `getIntrinsicElementProps` 😁